tools/dep_updaters/update-openssl.sh

```diff
@@ -10,13 +10,21 @@ cleanup() {
 }
 
 download() {
+
+  NEW_VERSION=$1
+
   if [ -z "$1" ]; then
-    echo "Error: please provide an OpenSSL version to update to"
-    echo "	e.g. ./$0 download 3.0.7+quic1"
-    exit 1
+      NEW_VERSION=$(gh api repos/quictls/openssl/releases -q '.[].tag_name|select(contains("openssl-3"))|ltrimstr("openssl-")' | head -n1)
```

Comment:
Does this script already have a dependency on `gh`? If not, maybe we should consider doing that call with something else (I think the other deps_updater scripts are using `node` for that).